integration_tests/e2e/goal/reviewUpdateGoal.cy.ts

```diff
@@ -18,6 +18,7 @@ context('Review updated goal', () => {
     cy.task('getPrisonerById')
     cy.task('stubGetInduction')
     cy.task('getActionPlan')
+    cy.task('getGoalsByStatus')
```

Comment:
Is the plan/approach that you're deprecating (removing) the calls to the `getActionPlan` stub and replacing with calls to `getGoalsByStatus`?
That appears to be what you're doing everywhere else, but not here? Does this mean that this cypress test needs both stubs setting up for some reason?

As a side note this is exactly why I'm struggling with this PR both in its sheer size, but also not knowing the design/approach you are taking here.
I'm guessing you've replaced `getActionPlan` calls on the Overview page with calls to the new `getGoalByStatus` API. But I wont know that until I get to the diff in the controller which is miles away! 🤣 
Normally we would put that kind of pertinent detail in some notes or narrative in the main PR description.

Reply:
Archive and update pages still use the action plan to load the specific goal. I think the best approach would be to add a "get goal" endpoint for those pages and remove getActionPlan entirely but that's obviously beyond the scope of this already large change.

Reply:
Ah, got it 👍 
Thanks 👍